vllm/v1/worker/gpu_input_batch.py

```diff
@@ -436,6 +471,13 @@ def _make_sampling_metadata(self) -> SamplingMetadata:
         else:
             prompt_token_ids = None
 
+        allowed_token_ids_mask: Optional[torch.Tensor] = None
+        if not self.no_allowed_token_ids and \
+                self.allowed_token_ids_mask is not None:
```

Comment:
I think no need for the second condition here since it will always have been allocated if the first condition is true.

Reply:
this is purely for type refinement. Otherwise, mypy will yell at me. I think it cannot tell the type of self. allowed_token_ids_mask[:num_req] if it doesn't know whether allowed_token_ids_mask is none or not.

Reply:
It's better in that case to `assert allowed_token_ids_mask is not None` within the `if`

Reply:
yup, good idea.